contrib/nsqd.go

```diff
@@ -0,0 +1,67 @@
+package contrib
+
+import (
+	"github.com/nsqio/nsq/internal/lg"
+	"github.com/nsqio/nsq/nsqd"
+	"strings"
+)
+
+type INSQD interface {
+	Logf(level lg.LogLevel, f string, args ...interface{})
+	GetStats() []nsqd.TopicStats
+	AddModuleGoroutine(addonFn func(exitChan chan int))
+}
+
+type initializer func([]string, INSQD) INSQDAddon
+
+type INSQDAddon interface {
+	Enabled() bool
+	Start()
+}
+
+type NSQDAddons struct {
+	addons []INSQDAddon
+}
+
+// Starts all addons that are active
+func (as *NSQDAddons) Start() {
+
+	for _, addon := range as.addons {
+		addon.Start()
+	}
+}
+
+// Initializes addons that have options set
+func NewEnabledNSQDAddons(contribOpts []string, n INSQD) *NSQDAddons {
+	var activeAddons []INSQDAddon
+	var hasOpt bool
+
+	initializers := map[string]initializer{
+		"dogstatsd": NewNSQDDogStatsd,
```

Comment:
@mreiferson if we used https://golang.org/pkg/plugin/ that functionality would go roughly here. We would scan for plugin files in some dir, open them and try to get two or three functions with standard names out.

The fact that the dogstatsd module is in the same "package" as this module loader makes it not match that pattern exactly - it has a unique init function name - but it's pretty close.